components/analytics-tools/org.wso2.carbon.analytics.tools.backup/src/main/java/org/wso2/carbon/analytics/tools/backup/AnalyticsDataBackupTool.java

```diff
@@ -320,5 +336,170 @@ private static AnalyticsSchema readTableSchema(String basePath) throws IOExcepti
+            if (analyticsFileSystem.length(node_path) == 0) { // the node is a directory
+                createDirectoryInLocalSystem(baseDir + File.separator + node_path);
+                backupFileSystemToLocal(analyticsFileSystem, node_path, baseDir);
+
+            } else {                                          // the node is a file
+                AnalyticsFileSystem.DataInput input = analyticsFileSystem.createInput(node_path);
+                byte[] dataInBuffer = new byte[READ_BUFFER_SIZE];
+                int i;
+                FileOutputStream out = new FileOutputStream(baseDir+node_path);
+                while ((i = input.read(dataInBuffer, 0, dataInBuffer.length)) > 0) {
+                    out.write(dataInBuffer);
+                }
+                out.close();
+            }
+        }
+    }
+
+    /**
+     * Restores the a local filesystem to the DAS filesystem
+     *
+     * @param analyticsFileSystem
+     * @param baseDir
+     * @throws IOException
+     */
+    private static void restoreFileSystem(AnalyticsFileSystem analyticsFileSystem, File baseDir)
+            throws IOException {
+        System.out.println("Restoring the file system with the Directory: " + baseDir);
+        restoreFileStructure(analyticsFileSystem, baseDir, baseDir);
+    }
+
+    /**
+     * Recursively travels through the filestructure and restores them
+     *
+     * @param analyticsFileSystem
+     * @param node
+     * @param baseDir
+     * @throws IOException
+     */
+    private static void restoreFileStructure(AnalyticsFileSystem analyticsFileSystem, File node, File baseDir) throws IOException {
+
+        //get the relative path
+        final String relativePath = node.getAbsolutePath().substring(baseDir.getParent().length());
+
+        if (node.isDirectory()) {
+            analyticsFileSystem.mkdir(relativePath);
+            String[] subNodes = node.list();
+            for (String filename : subNodes) {
+                restoreFileStructure(analyticsFileSystem, new File(node, filename), baseDir);
+            }
+        } else if (node.isFile()) {
+            OutputStream out = analyticsFileSystem.createOutput(relativePath);
+            byte[] data = readFile(node);
+            out.write(data, 0, data.length);
+            out.flush();
+            out.close();
+        }
+
+    }
+
+    /**
+     * Read the file into a byte array
+     *
+     * @param file
+     * @return
+     * @throws IOException
+     */
+    public static byte[] readFile(File file) throws IOException {
+
+        byte[] buffer = new byte[(int) file.length()];
+        InputStream inputStream = null;
+        try {
+            inputStream = new FileInputStream(file);
+            if (inputStream.read(buffer) == -1) {
+                throw new IOException("EOF reached while trying to read the whole file");
+            }
+        } finally {
+            try {
+                if (inputStream != null)
```

Comment:
This no longer applicable since this code got changed and refactored in a later PR.